src/primitives/transaction.cpp

```diff
@@ -36,7 +36,7 @@ std::string CTxIn::ToString() const
     if (prevout.IsNull())
         str += strprintf(", coinbase %s", HexStr(scriptSig));
     else
-        str += strprintf(", scriptSig=%s", scriptSig.ToString().substr(0,24));
+        str += strprintf(", scriptSig=%s", HexStr(scriptSig).substr(0, 24));
```

Comment:
was there a reason `HexStr` wasn't used here before?

Reply:
It was just more verbose (someone's personal preference, I guess), see what was removed `CScript::ToString`: https://github.com/bitcoin/bitcoin/commit/af3208bfa6967d6b35aecf0ba35d9d6bf0f8317e#diff-f7ca24fb80ddba0f291cb66344ca6fcb